File: _inc/laravel/app/Http/Controllers/Auth/PasswordResetLinkController.php
<?php

namespace App\Http\Controllers\Auth;

use App\Config\Constants\{SettingsConstants as SC, ViewsConstants as VW};
use App\Http\Controllers\Abstracts\Controller;
use App\Models\Utility;
use Illuminate\Http\{
    JsonResponse,
    RedirectResponse,
    Request,
    Response
};
use Illuminate\Support\Facades\{
    Log,
    Password,
    Validator
};
use Illuminate\View\View;
use function App\Http\Controllers\Helpers\{defaultUndefinedException, defaultPermissionDenial};

class PasswordResetLinkController extends Controller
{
    public const CRT = 'create';
    public const STR = 'store';


  public function create(Request $req): View|RedirectResponse|JsonResponse|null
  {
    $action = class_basename(static::class) . '@' . __FUNCTION__;
    return $this->measureProfile($action, function () use ($req, $action) {
      Log::info("{$action} – rendering email form view", ['view' => VW::PWD . 'email']);
      try {
        return view(VW::PWD . 'email');
      } catch (\Throwable $e) {
        Log::error("{$action} – exception thrown", [
            'file' => __FILE__,
            'class' => __CLASS__,
            'error_class' => get_class($e),
            'message' => $e->getMessage()
        ]);
        return self::_catch($req, $e);
      }
    });
  }

  public function store(Request $req): RedirectResponse|JsonResponse|null
  {
    $action = class_basename(static::class) . '@' . __FUNCTION__;
    return $this->measureProfile($action, function () use ($action, $req) {
      $valStart = microtime(true);
      if ($r = self::_validate($req)) return $r;
      $this->logExecutionTime($valStart, $action . '::_validate', 'completed');
      try {
        Log::info("[$action] Applying SMTP configuration", ['email' => $req->input('email')]);
        $smtpStart = microtime(true);
        Utility::smtpDetail(1);
        $this->logExecutionTime($smtpStart, $action . '::smtpDetail', 'completed');
        Log::info("[$action] Sending password reset link", ['email' => $req->input('email')]);
        $sendStart = microtime(true);
        $status = Password::sendResetLink($req->only('email'));
        $this->logExecutionTime($sendStart, $action . '::sendResetLink', 'completed');
        return $status === Password::RESET_LINK_SENT
          ? back()->with('status', __($status))
          : back()->withInput($req->only('email'))->withErrors(['email' => __($status)]);
      } catch (\Throwable $e) {
        Log::error("[$action] Error sending reset link", [
            'file' => __FILE__,
            'class' => __CLASS__,
            'error_class' => get_class($e),
            'message' => $e->getMessage()
        ]);
        return redirect()->back()->withErrors('E-Mail has not been sent due to SMTP configuration');
      }
    }, ['email' => $req->input('email')]);
  }

  private static function _catch(
    Request    $req,
    \Throwable $e
  ): RedirectResponse|JsonResponse|null {
    return defaultUndefinedException(
      $req,
      $e,
      __CLASS__ . '::' . debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 2)[1]['function']
    );
  }

  private static function _validate(
    Request $req
  ): RedirectResponse|JsonResponse|null {
    $rules = ['email' => 'required|email'];
    if (env('RECAPTCHA_MODULE') === 'on')
      $rules[SC::G_RCPT_RES] = 'required|captcha';
    $v = Validator::make($req->all(), $rules);
    return $v->fails()
      ? redirect()->back()->withErrors($v)->withInput()
      : null;
  }
}

File: _inc/_DEPRECATED_django/erp_prestech_backend/app/Http/Controllers/auth/password_reset_link_controller.py
import datetime
import json
import os
from django.conf import settings
from django.contrib import messages
from django.contrib.auth.forms import PasswordResetForm
from django.http import HttpRequest, HttpResponse
from django.shortcuts import redirect
from ....Models.utils.utility import Utility
from .._traits.controller import Controller

class PasswordResetLinkController(Controller):
  def get(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
    # TODO: Add additional logic if needed
    return HttpResponse(status=200)
  def post(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
    if getattr(settings, 'RECAPTCHA_MODULE', 'off') == 'on':
      recaptcha_response = request.POST.get('g-recaptcha-response')
      if not recaptcha_response:
        messages.error(request, "Please complete the reCAPTCHA.")
        return redirect('password_reset_link')
    email = request.POST.get('email')
    if not email:
      messages.error(request, "Email is required.")
      return redirect('password_reset_link')
    try:
      Utility.smtpDetail(1)
      form = PasswordResetForm({'email': email})
      messages.success(request, "Reset link sent.") if form.is_valid() else messages.error(request, "Error: " + str(form.errors))
      if form.is_valid():
        form.save(request=request, use_https=request.is_secure(), from_email=settings.EMAIL_HOST_USER)
      return redirect('password_reset_link')
    except Exception as e:
      # TODO: Add logging: Failed to POST for password reset: {e}
      messages.error(request, "E-Mail has not been sent due to SMTP configuration")
      return redirect('password_reset_link')
